### ISSUE TITLE
comment on issues too

### ISSUE THREAD

**Issue by zeke** [release-notifier/release-notifier#23]
Comments are currently created on Pull Requests that land in a release, but we should also be able to track down the source issues that are resolved by a release too.

cc @rsarky

**Comment by pvdlg** [comment]
For info we are parsing the body of commits and PR (found with GitHub search for the commits part of the release) to find issues fixed by those PR and commits in [semantic-release/github](https://github.com/semantic-release/github).

I couldn't find a solution that would parse properly the [GitHub closing keywords](https://help.github.com/articles/closing-issues-using-keywords). All the solution I found were "too naive" and missed some edge cases (ignoring references in back tick, fenced block or `<code>` blocks, missing weird variation with spaces, comma that GitHub recognize etc...). So I created [issue-parser](https://github.com/pvdlg/issue-parser).

It's mostly developed after lots and lots of empirical tests. It works well so far and I think most cases should be covered, but that would be great if we had a way to get the actual regex/algorithm used by GitHub.

Do you think it's something GitHub would open source? Or is there any plan for the GitHub API to add the issues closed in the metadata of the PR/commit retrieved via API?

**Comment by zeke** [comment]
>  So I created [issue-parser](https://github.com/pvdlg/issue-parser)

👍 

> would be great if we had a way to get the actual regex/algorithm used by GitHub.

cc @megbird, who appears to have written the [Closing issues using keywords](https://help.github.com/articles/closing-issues-using-keywords/) doc. Meagan, do you know much about the implementation of this?

**Comment by rsarky** [comment]
> would be great if we had a way to get the actual regex/algorithm used by GitHub.


I second this. I need a standardized way to parse a PR body for [helpr](https://github.com/rsarky/helpr). Currently it's very hacky and doesnt handle edge cases.

**Comment by pvdlg** [comment]
@rsarky you can check [issue-parser](https://github.com/pvdlg/issue-parser) in the meantime, I'll happily fix any edge cases you might find.

**Comment by rsarky** [comment]
@pvdlg Thanks :smile: I will check it out.

**Comment by zeke** [comment]
cc @cheshire137 who has done some work on this feature. Do you think extracting something reusable from GitHub's codebase is feasible?

**Comment by megbird** [comment]
:wave: sorry I didn't see this until today. @zeke and I spoke about the documentation history and about how our [move to nanoc](https://github.com/github/help-docs/commit/061881f79755cd8e4b7ae898d6a3d63c71b1465c) a few years ago means we can't see who worked on this when the feature was initially shipped. I wish I was more helpful 😸

**Comment by rsarky** [comment]
@zeke Would you be open to a PR on this that uses @pvdlg 's `issue-parser` . I have been using it on  [helPR](https://github.com/rsarky/helpr) and it's been working like a charm.
Thanks again @pvdlg :sparkles:

**Comment by zeke** [comment]
> Would you be open to a PR on this that uses @pvdlg 's issue-parser

@rsarky yes definitely!

**Comment by rsarky** [comment]
So basically for each PR that was present in a release : 
- Parse the PR body using `issue-parser`.
- If any issues are referenced add a comment saying they were fixed in so and so release.

I was wondering if this would be helpful though.
Issues automatically get closed whenever a PR referencing it gets merged. Is commenting on a closed issue gonna be useful?

**Comment by zeke** [comment]
> Is commenting on a closed issue gonna be useful?

Yes, as the author and commenters will still be subscribed to it even though it's closed, and they may not be subscribed to updates on the resolving PR.

**Comment by rsarky** [comment]
@zeke That makes sense. I have my exams going on .
I will try to submit a PR by the weekend .
